### PR TITLE
[core]control the output format of access.log through the http.access_log_format field in config.yaml

### PR DESCRIPTION
**Pull request by Miss-you**
You can control the output format of access.log through the http.access_log_format field in config.yaml, and add a recommended access.log format in the comments.

NOTE: Please read the Contributing.md guidelines before submitting your patch:

https://github.com/apache/incubator-apisix/blob/master/Contributing.md#how-to-add-a-new-feature-or-change-an-existing-one

### Summary

SUMMARY_HERE

### Full changelog

* [Implement ...]
* [Add related tests]
* ...

### Issues resolved

Fix #XXX

### CODE REVIEW

**Comment by Miss-you** [comment]
Need the test cases and documentation

File: conf/config.yaml [hunk 1]
```diff
@@ -100,6 +100,12 @@ nginx_config:                     # config for render the template to genarate n
     worker_connections: 10620
   http:
     access_log: "logs/access.log"
+    #access_log_format: |+         # access.log recommended format
+    #  '{"host":"$http_host", "uri":"$uri", "scheme":"$scheme", "rsp_st":"$status", "ups_st":"$upstream_status", "cip":"$remote_addr", '
```
Comment (membphis) [inline conf/config.yaml:104]: we need to add `escape=json`:

http://nginx.org/en/docs/http/ngx_http_log_module.html#log_format

File: conf/config.yaml [hunk 2]
```diff
@@ -100,6 +100,12 @@ nginx_config:                     # config for render the template to genarate n
     worker_connections: 10620
   http:
     access_log: "logs/access.log"
+    #access_log_format: |+         # access.log recommended format
```
Comment (membphis) [inline conf/config.yaml:103]: `access.log recommended format` -> `access.log recommended json format`

**Comment by membphis** [comment]
The current test case is not enough.

We need to read the last line of `access.log`, and confirm it is a valid JSON string.

File: .travis/apisix_cli_test.sh [hunk 3]
```diff
@@ -72,3 +72,19 @@ done
...
 sed -i '/dns_resolver:/,+4s/^#//'  conf/config.yaml
 echo "passed: system nameserver imported"
+
+# check access log format
+
+sed -i 's/#access_log_format: |+/access_log_format: "test config accesslog"/g' conf/config.yaml
+
+make init
+
+grep "log_format main test config accesslog" conf/nginx.conf > /dev/null
```
Comment (moonming) [inline .travis/apisix_cli_test.sh:82]: this test case is not enough.

**Comment by membphis** [comment]
@Miss-you any news?

**Comment by moonming** [comment]
this PR that hasn't been updated for a long time, close it first

**Pull request closed without merging**